cache/bug_cache.go

```diff
@@ -78,6 +96,9 @@ func (c *BugCache) AddCommentWithFiles(message string, files []repository.Hash)
 }
 
 func (c *BugCache) AddCommentRaw(author *IdentityCache, unixTime int64, message string, files []repository.Hash, metadata map[string]string) (*bug.AddCommentOperation, error) {
+	c.Lock()
```

Comment:
If the locking is done externally with `b.Lock()` and `b.Unlock()`, you should not write lock here anymore.

You need to keep the read locks though.